src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -382,13 +411,39 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
 
         aMessage.Read(offset, sizeof(Tlv), &data.tlv);
         aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
-        mLocal.Set(data.tlv);
+
+        if (!isUpdateFromCommissioner && isTlvsAffectConnectivity)
+        {
+            mNetif.GetPendingDataset().GetNetwork().Set(data.tlv);;
```

Comment:
When a Thread device submits an Active Dataset that affects connectivity, we need to:

1. Completely replace the Pending Dataset with the newly received Active Dataset.
2. Add a Delay Timer TLV with the MIN_DELAY_TIMER value.
3. Add a Pending Dataset Timestamp TLV.

When the Delay Timer expires, the Active Dataset should be exactly what was originally received from the Thread device.